### PR TITLE
Result patch

### PR DESCRIPTION
**Pull request by smidf**
- 🔥 The `ApiErrorResult` has been removed and `ErrorResult` is final.

### CODE REVIEW

File: kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt [hunk 1]
```diff
@@ -71,13 +71,19 @@ interface ErrorCode {
...
+    override fun hashCode(): Int = value.hashCode()
+
+    override fun toString(): String = value.toString()
 }
 
-open class ErrorResult(
-    open val code: ErrorCode = ErrorCode.UNDEFINED,
-    open val message: String? = null,
-    open val throwable: Throwable? = null
+data class ErrorResult(
```
Comment (Del-S) [inline kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt:83]: Not really sure that `ErrorResult` should be final. In the CSOB app we extend it to modify code variable type and add some additional values.

**Review by Del-S — CHANGES_REQUESTED** [review]
Check the comments please.

File: kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt (hunk 1 above)
Comment (vsouhrada) [reply kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt:83]: Why is @smidf  final? This class is extend it very often. Specific process has own error result implementation based on this base class

Comment (smidf) [reply kaal-domain/src/main/kotlin/cz/eman/kaal/domain/result/ExecutionContext.kt:77]: Yes, I think the `ErrorResult` extending is anti-pattern. The `Error` class / state is something, that describes the error, and the code (unique identifier), the message and the stacktrace are sufficient. If a user has the ability to extend, add, or override class attributes, he can exploit the error to pass data.

Only `ErrorCode` should be domain specific, not `ErrorResult`.

**Comment by PistaToth** [comment]
Domluvili jsme se ze to zavreme. Filip ma aktualnejsi reseni.

**Pull request closed without merging**